presto-main/src/main/java/com/facebook/presto/cost/ScalarStatsCalculator.java

```diff
@@ -91,5 +96,62 @@ protected SymbolStatsEstimate visitLiteral(Literal node, Void context)
                     .setDistinctValuesCount(1)
                     .build();
         }
+
+        @Override
+        protected SymbolStatsEstimate visitArithmeticBinary(ArithmeticBinaryExpression node, Void context)
+        {
+            requireNonNull(node, "node is null");
+            checkArgument(node.getType() != ArithmeticBinaryExpression.Type.MODULUS, "Modulus operator not yet supported");
```

Comment:
You cannot throw here. It will fail query.
instead return `UNKNOWN_STATS`.

Either explicitly or by returning 
`super.visitArithmeticBinary(node, context)`

Edit:
Actually you can compute some stats even without understanding MODULUS arithmetic. You should be good with NullsFraction and DataSize.

And you can cap `numberDistinctValues` in result by `rightHigh` (if we are in integer domain) and always by `leftHigh`